File: ios/Talise/Features/Payroll/TeamStreamSetupView.swift
import SwiftUI

/// Set up a streaming payout for a team: fund a pot once, then equal shares
/// stream to every member on an interval, gaslessly, until the pot runs out.
///
/// Flow: createPrepare (draft + escrow address) → fund the escrow over the normal
/// gasless send rail (signAndSubmitSend) → record (activate). A backend cron then
/// releases each tranche. No per-tranche signing by the user.
struct TeamStreamSetupView: View {
    let team: TeamDTO
    @Environment(\.dismiss) private var dismiss
    @Environment(AppSession.self) private var session

    @State private var amount: String = ""
    @State private var numTranches: Int = 4
    @State private var interval: Interval = .daily
    @State private var starting = false
    @State private var error: String?
    @State private var resetSlider = false
    @State private var started = false
    @State private var startedSummary: String = ""

    /// Payout cadence presets → minutes.
    enum Interval: String, CaseIterable, Identifiable {
        case minute = "Every minute"
        case hourly = "Hourly"
        case daily = "Daily"
        case weekly = "Weekly"
        var id: String { rawValue }
        var minutes: Int { switch self { case .minute: 1; case .hourly: 60; case .daily: 1440; case .weekly: 10080 } }
        var unit: String { switch self { case .minute: "minute"; case .hourly: "hour"; case .daily: "day"; case .weekly: "week" } }
    }

    private var memberCount: Int { max(team.members.count, 1) }
    private var totalUsd: Double { Double(amount.trimmingCharacters(in: .whitespaces)) ?? 0 }
    private var perMemberPerPayout: Double {
        guard numTranches > 0 else { return 0 }
        return totalUsd / Double(numTranches) / Double(memberCount)
    }
    private var canStart: Bool { totalUsd > 0 && numTranches >= 1 && perMemberPerPayout >= 0.01 && !starting }

    var body: some View {
        if started { successView } else { form }
    }

    // MARK: - Form

    private var form: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 18) {
                header
                amountCard
                scheduleCard
                summaryCard

                if let error {
                    Text(error)
                        .font(TaliseFont.body(13, weight: .light))
                        .foregroundStyle(TaliseColor.danger)
                        .fixedSize(horizontal: false, vertical: true)
                }

                SlideToConfirm(title: starting ? "Starting…" : "Slide to start streaming",
                               tint: TaliseColor.accent,
                               reset: $resetSlider) {
                    await start()
                }
                .disabled(!canStart)
                .opacity(canStart ? 1 : 0.6)

                Text("One gasless transaction funds the pot. Payouts release automatically — no gas, ever.")
                    .font(TaliseFont.mono(11, weight: .regular))
                    .foregroundStyle(TaliseColor.fgMuted)

                Color.clear.frame(height: 24)
            }
            .padding(.horizontal, 20)
            .padding(.top, 8)
        }
        .background(TaliseColor.bg.ignoresSafeArea())
        .navigationTitle("")
        .navigationBarTitleDisplayMode(.inline)
        .scrollDismissesKeyboard(.interactively)
    }

    private var header: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text("STREAM TO TEAM")
                .font(TaliseFont.mono(10, weight: .regular)).kerning(1.4)
                .foregroundStyle(TaliseColor.fgDim)
            Text("Stream to \(team.name)")
                .font(TaliseFont.heading(24, weight: .medium)).kerning(-0.5)
                .foregroundStyle(TaliseColor.fg)
            Text("Fund a pot once. Everyone gets an equal share on a schedule — automatically and gaslessly.")
                .font(TaliseFont.body(13, weight: .light))
                .foregroundStyle(TaliseColor.fgMuted)
                .fixedSize(horizontal: false, vertical: true)
        }
        .padding(.top, 4)
    }

    private var amountCard: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text("AMOUNT TO STREAM")
                .font(TaliseFont.mono(10, weight: .regular)).kerning(0.6)
                .foregroundStyle(TaliseColor.fgDim)
            HStack(spacing: 4) {
                Text("$").font(TaliseFont.heading(22, weight: .medium)).foregroundStyle(TaliseColor.fgMuted)
                TextField("", text: $amount, prompt: Text("10.00").foregroundColor(TaliseColor.fgDim))
                    .font(TaliseFont.heading(22, weight: .medium))
                    .foregroundStyle(TaliseColor.fg)
                    .keyboardType(.decimalPad)
            }
            .padding(.horizontal, 14).frame(height: 54)
            .background(RoundedRectangle(cornerRadius: 14, style: .continuous).fill(TaliseColor.surface2))
        }
        .padding(16)
        .rampCard()
    }

    private var scheduleCard: some View {
        VStack(alignment: .leading, spacing: 14) {
            HStack {
                Text("PAYOUTS")
                    .font(TaliseFont.mono(10, weight: .regular)).kerning(0.6)
                    .foregroundStyle(TaliseColor.fgDim)
                Spacer()
                Stepper(value: $numTranches, in: 1...365) {
                    Text("\(numTranches)")
                        .font(TaliseFont.heading(16, weight: .medium))
                        .foregroundStyle(TaliseColor.fg)
                }
                .labelsHidden()
                .fixedSize()
            }
            Divider().overlay(TaliseColor.line)
            HStack {
                Text("HOW OFTEN")
                    .font(TaliseFont.mono(10, weight: .regular)).kerning(0.6)
                    .foregroundStyle(TaliseColor.fgDim)
                Spacer()
                Picker("", selection: $interval) {
                    ForEach(Interval.allCases) { Text($0.rawValue).tag($0) }
                }
                .pickerStyle(.menu)
                .tint(TaliseColor.accent)
            }
        }
        .padding(16)
        .rampCard()
    }

    private var summaryCard: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text("PREVIEW")
                .font(TaliseFont.mono(10, weight: .regular)).kerning(0.6)
                .foregroundStyle(TaliseColor.fgDim)
            Text("\(memberCount) \(memberCount == 1 ? "person" : "people") each get \(TaliseFormat.usd2(perMemberPerPayout)) per \(interval.unit)")
                .font(TaliseFont.body(14, weight: .regular))
                .foregroundStyle(TaliseColor.fg)
                .fixedSize(horizontal: false, vertical: true)
            Text("\(numTranches) payouts · \(TaliseFormat.usd2(totalUsd)) total")
                .font(TaliseFont.mono(11, weight: .regular))
                .foregroundStyle(TaliseColor.fgMuted)
            if totalUsd > 0 && perMemberPerPayout < 0.01 {
                Text("Each share is too small — add more or use fewer payouts (min $0.01 each).")
                    .font(TaliseFont.body(12, weight: .light))
                    .foregroundStyle(TaliseColor.danger)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .rampCard()
    }

    // MARK: - Success

    private var successView: some View {
        VStack(spacing: 16) {
            Spacer()
            ZStack {
                Circle().fill(TaliseColor.accent.opacity(0.16)).frame(width: 92, height: 92)
                Image(systemName: "dot.radiowaves.left.and.right")
                    .font(.system(size: 40, weight: .regular))
                    .foregroundStyle(TaliseColor.accent)
            }
            Text("Streaming started")
                .font(TaliseFont.heading(24, weight: .medium))
                .foregroundStyle(TaliseColor.fg)
            Text(startedSummary)
                .font(TaliseFont.body(14, weight: .light))
                .foregroundStyle(TaliseColor.fgMuted)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 28)
            Spacer()
            Button { dismiss() } label: {
                Text("Done")
                    .font(TaliseFont.body(16, weight: .semibold)).foregroundStyle(.black)
                    .frame(maxWidth: .infinity).frame(height: 54)
                    .background(Capsule().fill(TaliseColor.greenMint))
            }
            .buttonStyle(.plain)
            .padding(.horizontal, 20).padding(.bottom, 24)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(TaliseColor.bg.ignoresSafeArea())
    }

    // MARK: - Action

    private func start() async {
        guard canStart else { resetSlider.toggle(); return }
        starting = true; error = nil
        defer { starting = false }
        do {
            // 1) Draft + escrow address.
            let prep = try await TeamStreamAPI.createPrepare(
                teamId: team.id,
                totalUsd: totalUsd,
                numTranches: numTranches,
                intervalMinutes: interval.minutes
            )
            // 2) Fund the escrow over the normal gasless send rail.
            let sub = try await ZkLoginCoordinator.shared.signAndSubmitSend(
                to: prep.escrowAddress,
                amountUsd: prep.totalUsd,
                intent: "Fund team stream"
            )
            // 3) Activate the stream with the funding digest.
            _ = try await TeamStreamAPI.record(streamId: prep.streamId, digest: sub.digest)

            startedSummary = "\(prep.memberCount) \(prep.memberCount == 1 ? "person" : "people") will each receive \(TaliseFormat.usd2(prep.perMemberUsd)) per \(interval.unit), \(prep.numTranches) times."
            NotificationCenter.default.post(name: .taliseTxCompleted, object: TaliseTxEvent(
                digest: sub.digest, direction: "sent", amountUsdsui: prep.totalUsd,
                counterparty: nil, counterpartyName: team.name, venue: nil))
            withAnimation { started = true }
        } catch ZkLoginCoordinator.SessionError.rebindRequired {
            error = "Sign in again — your session needs a refresh."
            resetSlider.toggle()
            session.signOut()
        } catch {
            if APIError.isCancellation(error) { return }
            self.error = APIError.honestMoneyError(error, fallback: "Couldn't start the stream. Please try again.")
            resetSlider.toggle()
        }
    }
}

File: android/app/src/main/java/io/talise/app/feature/payroll/TeamStreamSetupView.kt
package io.talise.app.feature.payroll

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.UnfoldMore
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewModelScope
import androidx.lifecycle.viewmodel.compose.viewModel
import io.talise.app.core.model.TeamDTO
import io.talise.app.ui.components.SlideToConfirm
import io.talise.app.ui.components.rampCard
import io.talise.app.ui.theme.TaliseColors
import io.talise.app.ui.theme.TaliseType
import kotlinx.coroutines.CancellationException
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

/**
 * Set up a streaming payout for a team, a 1:1 port of iOS TeamStreamSetupView.
 * Fund a pot once, then equal shares stream to every member on an interval,
 * gaslessly, until the pot runs out.
 *
 * Flow: streamCreatePrepare (draft + escrow address), fund the escrow over the
 * normal gasless send rail (signAndSubmitSend), then streamRecord (activate).
 * A backend cron releases each tranche, no per-tranche signing by the user.
 */

private enum class StreamInterval(val label: String, val minutes: Int, val unit: String) {
    Minute("Every minute", 1, "minute"),
    Hourly("Hourly", 60, "hour"),
    Daily("Daily", 1440, "day"),
    Weekly("Weekly", 10080, "week"),
}

/**
 * Owns the fund-the-pot pipeline (prepare -> sign+send -> record) so it runs on
 * [viewModelScope] and survives the composable. On the previous
 * `rememberCoroutineScope` a back press mid-flight cancelled the coroutine
 * between `signAndSubmitSend` (real USDsui moved to escrow) and `streamRecord`
 * (activation), silently stranding the funds.
 */
class TeamStreamSetupViewModel : ViewModel() {

    data class State(
        val starting: Boolean = false,
        val error: String? = null,
        val started: Boolean = false,
        val startedSummary: String = "",
        /** Forces the slider knob back to start after a failed confirm. */
        val resetSliderTick: Int = 0,
    )

    private val _state = MutableStateFlow(State())
    val state: StateFlow<State> = _state.asStateFlow()

    fun bounceSlider() {
        _state.value = _state.value.copy(resetSliderTick = _state.value.resetSliderTick + 1)
    }

    fun start(teamId: String, totalUsd: Double, numTranches: Int, intervalMinutes: Int, intervalUnit: String) {
        if (_state.value.starting) return
        _state.value = _state.value.copy(starting = true, error = null)
        viewModelScope.launch {
            try {
                val prep = PayrollApi.service.streamCreatePrepare(
                    StreamCreateBody(
                        teamId = teamId,
                        totalUsd = totalUsd,
                        numTranches = numTranches,
                        intervalMinutes = intervalMinutes,
                    ),
                )
                val digest = PayrollApi.signAndSubmitSend(to = prep.escrowAddress, amountUsd = prep.totalUsd)
                PayrollApi.service.streamRecord(StreamRecordBody(streamId = prep.streamId, digest = digest))
                val people = if (prep.memberCount == 1) "person" else "people"
                _state.value = _state.value.copy(
                    starting = false,
                    started = true,
                    startedSummary = "${prep.memberCount} $people will each receive ${payrollUsd2(prep.perMemberUsd)} per $intervalUnit, ${prep.numTranches} times.",
                )
            } catch (e: CancellationException) {
                throw e
            } catch (e: Exception) {
                _state.value = _state.value.copy(
                    starting = false,
                    error = PayrollApi.friendlyPayoutError(e, "Couldn't start the stream. Please try again."),
                    resetSliderTick = _state.value.resetSliderTick + 1,
                )
            }
        }
    }
}

@Composable
fun TeamStreamSetupView(
    team: TeamDTO,
    onDismiss: () -> Unit,
    vm: TeamStreamSetupViewModel = viewModel(key = "team-stream-${team.id}") { TeamStreamSetupViewModel() },
) {
    var amount by remember { mutableStateOf("") }
    var numTranches by remember { mutableStateOf(4) }
    var interval by remember { mutableStateOf(StreamInterval.Daily) }
    val state by vm.state.collectAsStateWithLifecycle()
    val starting = state.starting

    // Tick -> one-frame reset pulse for the slider (same wiring as PayTeamView).
    var resetSlider by remember { mutableStateOf(false) }
    LaunchedEffect(state.resetSliderTick) {
        if (state.resetSliderTick > 0) {
            resetSlider = true
            delay(50)
            resetSlider = false
        }
    }

    val memberCount = maxOf(team.members.size, 1)
    val totalUsd = amount.trim().toDoubleOrNull() ?: 0.0
    val perMemberPerPayout = if (numTranches > 0) totalUsd / numTranches / memberCount else 0.0
    val canStart = totalUsd > 0 && numTranches >= 1 && perMemberPerPayout >= 0.01 && !starting

    fun start() {
        if (!canStart) { vm.bounceSlider(); return }
        vm.start(
            teamId = team.id,
            totalUsd = totalUsd,
            numTranches = numTranches,
            intervalMinutes = interval.minutes,
            intervalUnit = interval.unit,
        )
    }

    if (state.started) {
        StreamStartedView(summary = state.startedSummary, onDone = onDismiss)
        return
    }

    Column(
        Modifier.fillMaxSize().background(TaliseColors.bg).verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp).padding(top = 8.dp),
        verticalArrangement = Arrangement.spacedBy(18.dp),
    ) {
        // Header
        Column(Modifier.padding(top = 4.dp), verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text("STREAM TO TEAM", style = TaliseType.mono(10.sp), letterSpacing = 1.4.sp, color = TaliseColors.fgDim)
            Text("Stream to ${team.name}", style = TaliseType.heading(24.sp, FontWeight.Medium), letterSpacing = (-0.5).sp, color = TaliseColors.fg)
            Text(
                "Fund a pot once. Everyone gets an equal share on a schedule, automatically and gaslessly.",
                style = TaliseType.body(13.sp, FontWeight.Light), color = TaliseColors.fgMuted,
            )
        }

        // Amount card
        Column(Modifier.fillMaxWidth().rampCard().padding(16.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
            Text("AMOUNT TO STREAM", style = TaliseType.mono(10.sp), letterSpacing = 0.6.sp, color = TaliseColors.fgDim)
            Row(
                Modifier.fillMaxWidth().height(54.dp).background(TaliseColors.surface2, RoundedCornerShape(14.dp))
                    .padding(horizontal = 14.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp),
            ) {
                Text("$", style = TaliseType.heading(22.sp, FontWeight.Medium), color = TaliseColors.fgMuted)
                TextField(
                    value = amount,
                    onValueChange = { s -> amount = s.filter { it.isDigit() || it == '.' } },
                    modifier = Modifier.fillMaxWidth(),
                    placeholder = { Text("10.00", style = TaliseType.heading(22.sp, FontWeight.Medium), color = TaliseColors.fgDim) },
                    textStyle = TaliseType.heading(22.sp, FontWeight.Medium).copy(color = TaliseColors.fg),
                    singleLine = true,
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
                    colors = transparentFieldColors(),
                )
            }
        }

        // Schedule card
        Column(Modifier.fillMaxWidth().rampCard().padding(16.dp), verticalArrangement = Arrangement.spacedBy(14.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("PAYOUTS", style = TaliseType.mono(10.sp), letterSpacing = 0.6.sp, color = TaliseColors.fgDim)
                Spacer(Modifier.weight(1f))
                StepperControl(
                    value = numTranches,
                    onDec = { if (numTranches > 1) numTranches -= 1 },
                    onInc = { if (numTranches < 365) numTranches += 1 },
                )
            }
            Box(Modifier.fillMaxWidth().height(1.dp).background(TaliseColors.line))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("HOW OFTEN", style = TaliseType.mono(10.sp), letterSpacing = 0.6.sp, color = TaliseColors.fgDim)
                Spacer(Modifier.weight(1f))
                IntervalPicker(selected = interval, onSelect = { interval = it })
            }
        }

        // Summary card
        Column(Modifier.fillMaxWidth().rampCard().padding(16.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
            Text("PREVIEW", style = TaliseType.mono(10.sp), letterSpacing = 0.6.sp, color = TaliseColors.fgDim)
            val people = if (memberCount == 1) "person" else "people"
            Text(
                "$memberCount $people each get ${payrollUsd2(perMemberPerPayout)} per ${interval.unit}",
                style = TaliseType.body(14.sp, FontWeight.Normal), color = TaliseColors.fg,
            )
            Text("$numTranches payouts, ${payrollUsd2(totalUsd)} total", style = TaliseType.mono(11.sp), color = TaliseColors.fgMuted)
            if (totalUsd > 0 && perMemberPerPayout < 0.01) {
                Text(
                    "Each share is too small, add more or use fewer payouts (min $0.01 each).",
                    style = TaliseType.body(12.sp, FontWeight.Light), color = TaliseColors.danger,
                )
            }
        }

        state.error?.let {
            Text(it, style = TaliseType.body(13.sp, FontWeight.Light), color = TaliseColors.danger)
        }

        SlideToConfirm(
            title = if (starting) "Starting…" else "Slide to start streaming",
            onConfirm = { start() },
            enabled = canStart,
            tint = TaliseColors.accent,
            reset = resetSlider,
        )

        Text(
            "One gasless transaction funds the pot. Payouts release automatically, no gas, ever.",
            style = TaliseType.mono(11.sp), color = TaliseColors.fgMuted,
        )
        Spacer(Modifier.height(24.dp))
    }
}

@Composable
private fun StepperControl(value: Int, onDec: () -> Unit, onInc: () -> Unit) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp)) {
        StepperButton(Icons.Filled.Remove, onDec)
        Text("$value", style = TaliseType.heading(16.sp, FontWeight.Medium), color = TaliseColors.fg)
        StepperButton(Icons.Filled.Add, onInc)
    }
}

@Composable
private fun StepperButton(icon: androidx.compose.ui.graphics.vector.ImageVector, onClick: () -> Unit) {
    Box(
        Modifier.size(30.dp).background(TaliseColors.surface2, CircleShape).clickable { onClick() },
        contentAlignment = Alignment.Center,
    ) { Icon(icon, null, tint = TaliseColors.fg, modifier = Modifier.size(16.dp)) }
}

@Composable
private fun IntervalPicker(selected: StreamInterval, onSelect: (StreamInterval) -> Unit) {
    var open by remember { mutableStateOf(false) }
    Box {
        Row(
            Modifier.clickable { open = true }.padding(horizontal = 8.dp, vertical = 4.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Text(selected.label, style = TaliseType.body(15.sp, FontWeight.Medium), color = TaliseColors.accent)
            Icon(Icons.Filled.UnfoldMore, null, tint = TaliseColors.accent, modifier = Modifier.size(16.dp))
        }
        DropdownMenu(expanded = open, onDismissRequest = { open = false }) {
            StreamInterval.entries.forEach { opt ->
                DropdownMenuItem(
                    text = { Text(opt.label, style = TaliseType.body(15.sp, FontWeight.Normal), color = TaliseColors.fg) },
                    onClick = { onSelect(opt); open = false },
                )
            }
        }
    }
}

@Composable
private fun StreamStartedView(summary: String, onDone: () -> Unit) {
    Column(
        Modifier.fillMaxSize().background(TaliseColors.bg).padding(horizontal = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Box(Modifier.size(92.dp).background(TaliseColors.accent.copy(alpha = 0.16f), CircleShape), contentAlignment = Alignment.Center) {
            Icon(Icons.Filled.UnfoldMore, null, tint = TaliseColors.accent, modifier = Modifier.size(40.dp))
        }
        Spacer(Modifier.height(16.dp))
        Text("Streaming started", style = TaliseType.heading(24.sp, FontWeight.Medium), color = TaliseColors.fg)
        Spacer(Modifier.height(8.dp))
        Text(summary, style = TaliseType.body(14.sp, FontWeight.Light), color = TaliseColors.fgMuted)
        Spacer(Modifier.height(28.dp))
        Box(
            Modifier.fillMaxWidth().height(54.dp).background(TaliseColors.greenMint, RoundedCornerShape(27.dp)).clickable { onDone() },
            contentAlignment = Alignment.Center,
        ) { Text("Done", style = TaliseType.body(16.sp, FontWeight.SemiBold), color = TaliseColors.inkOnGreen) }
    }
}

@Composable
private fun transparentFieldColors() = TextFieldDefaults.colors(
    focusedContainerColor = Color.Transparent,
    unfocusedContainerColor = Color.Transparent,
    disabledContainerColor = Color.Transparent,
    focusedIndicatorColor = Color.Transparent,
    unfocusedIndicatorColor = Color.Transparent,
    disabledIndicatorColor = Color.Transparent,
    cursorColor = TaliseColors.accent,
)
